File: 1.两数之和/main.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    var map = <int, int>{};
    for (int i = 0; i < nums.length; ++i) {
      var diff = target - nums[i];

      if (map[diff] != null) {
        return [map[diff]!, i];
      }

      map[nums[i]] = i;
    }

    return [0, 0];
  }
}

File: 1.两数之和/main.ts
function twoSum(nums: number[], target: number): number[] {
  const m: Map<number, number> = new Map();

  for (let i = 0; i < nums.length; ++i) {
    const diff = target - nums[i];

    if (m.has(diff)) {
      return [m.get(diff)!, i];
    }
    m.set(nums[i], i);
  }

  throw new Error();
}
